src/components/Button/src/Button.vue

```diff
@@ -189,7 +253,7 @@ export default {
 		variant: {
 			type: String,
 			default: 'primary',
-			validator: (variant) => ['primary', 'secondary', 'tertiary'].includes(variant),
+			validator: (variant) => ['primary', 'secondary', 'tertiary', 'success', 'error'].includes(variant),
```

Comment:
We might want to consider making these constants at some point if we're going to repeat the strings everywhere

Reply:
The string literals are already their own constants. Turning them into JS constants makes the code more verbose with no benefit:

```js
// some/constants/file.js

export const SUCCESS = 'success';
export const ERROR = 'error';

// some/vue/component.vue

import { SUCCESS, ERROR } from 'some/constants/file.js';
// ...
validator(variant) => [SUCCESS, ERROR].includes(variant),
```

vs just keeping it:

```js
// some/vue/component.vue

validator(variant) => ['success', 'error'].includes(variant),
```

It's "easier to refactor" the previous version _but not really_. For example, if we had:
```js
export const ERROR = 'error';
```

And we wanted to rename it to `'failure'` we wouldn't just do this:
```js
export const ERROR = 'failure';
```
And leave it at that, because now it's confusing that `ERROR = 'failure'` and not `ERROR = 'error'` in the rest of the codebase, so we'd much more likely do this instead:
```js
export const FAILURE = 'failure';
```
Which ironically doesn't actually save us any work from just using string literals in the first place because now we have to go over the entire codebase and change `ERROR` into `FAILURE` anyway.

Furthermore, if the constants are used in the public API of the components then changing them is a breaking change and requires a major version release so people very rarely, if ever, change them because it's often not worth the hassle.

I think it's okay for us to stick to using string literals as their own constants for now. I'm willing to revisit this idea in the future if it can be shown it provides a benefit that's actually greater than its maintenance cost.

Reply:
Works for me!

Reply:
I've always been on the fence about constants. I agree it doesn't help with refactoring. But, I also thought constants help catch typos.
```js
if (variant === SUCCES) // unknown variable error
if (variant === 'succes') // harder to spot bug
```
That being said, it's a lot of overhead.